frontend/src/store/index.js

```diff
@@ -171,18 +171,23 @@ const getters = {
       const allMachineImages = flatMap(machineImages, machineImage => {
         const machineImageVersions = []
         forEach(machineImage.versions, version => {
-          if (!version.expirationDate || moment().isBefore(version.expirationDate)) {
-            const vendorName = vendorNameFromImageName(machineImage.name)
-            machineImageVersions.push({
-              name: machineImage.name,
-              version: version.version,
-              expirationDate: version.expirationDate,
-              expirationDateString: getDateFormatted(version.expirationDate),
-              vendorName,
-              icon: iconForVendor(vendorName),
-              needsLicense: vendorNeedsLicense(vendorName)
-            })
+          if (version.expirationDate && moment().isBefore(version.expirationDate)) {
```

Comment:
replace forEach with filter / map and sort in sequence for better readability.
maybe introduce a function for second flatMap argument.

Reply:
done